unity-renderer/Assets/UIComponents/Scripts/Components/BaseComponentView.cs

```diff
@@ -184,4 +154,24 @@ internal IEnumerator RaiseOnScreenSizeChangedAfterDelay()
         T buttonComponentView = Instantiate(Resources.Load<GameObject>(resourceName)).GetComponent<T>();
         return buttonComponentView;
     }
-}
\ No newline at end of file
+}
+
+public interface IBaseComponentView<in TModel> : IBaseComponentView
+{
+    void SetModel(TModel newModel);
+}
+
+public abstract class BaseComponentView<TModel> : BaseComponentView, IBaseComponentView<TModel> where TModel : IEquatable<TModel>
```

Comment:
would be nice to have some namespace ;)

Reply:
I will this class only to the namespace as I don't want to make too much git noise. We may consider to distribute between namespaces automatically in `dev`

Reply:
Since it could be potentially a "breaking change", I would not do it in the `dev`, but rather inside PRs. However, it could be probably done automatically if it is aligned with the folder structure. And we agreed on that with the team (just that namesapces would not go deep hierarchy, more flat). But it will require folder structure re-organization...